autoemulate/emulators/transformed/base.py

```diff
@@ -392,21 +401,69 @@ def _predict(self, x: TensorLike, with_grad: bool) -> OutputLike:
         if not self.output_from_samples:
             if isinstance(y_t_pred, GaussianLike):
                 return self._inv_transform_y_gaussian(y_t_pred)
-            if isinstance(y_t_pred, DistributionLike):
-                return self._inv_transform_y_distribution(y_t_pred)
-            msg = "y_pred is not TensorLike, GaussianLike or DistributionLike"
+            msg = (
+                f"y_t_pred ({type(y_t_pred)}) is not `GaussianLike` and not currently "
+                "supported for when inverting without sampling."
+            )
             raise ValueError(msg)
 
         # Output derived by sampling and inverting to original space
         if isinstance(y_t_pred, DistributionLike):
-            return self._inv_transform_y_gaussian_sample(y_t_pred)
+            y_pred = self._inv_transform_y_distribution(y_t_pred)
+            samples = y_pred.rsample(torch.Size([self.n_samples]))
+            if not self.full_covariance:
+                return GaussianLike(
+                    samples.mean(dim=0),
+                    DiagLinearOperator(samples.var(dim=0, unbiased=False)),
+                )
+            msg = (
+                "Full covariance sampling is not currently implemented for "
+                "`TransformedEmulator` with sampling-based predictions since a "
+                "consistent structure for the covariance matrix cannot be guranteed "
+                "for any set of composed `torch.distributions.Transform`."
+            )
+            raise NotImplementedError(msg)
+
         msg = (
             "Invalid output type from model prediction. Expected TensorLike,"
             "GaussianLike, or DistributionLike. Received: "
             f"{type(y_t_pred)}"
         )
         raise ValueError(msg)
 
+    def predict_mean_and_variance(
+        self, x: TensorLike, with_grad: bool = False
+    ) -> tuple[TensorLike, TensorLike]:
+        """
+        Predict the mean and variance of the target variable for input `x`.
+
+        Parameters
+        ----------
+        x: TensorLike
+            Input tensor of shape `(n_samples, n_features)` for which to predict
+            the mean and variance.
+        with_grad: bool
+            Whether to compute gradients with respect to the input. Defaults to False.
+
+        Returns
+        -------
+        tuple[TensorLike, TensorLike]
+            A tuple containing:
+            - Mean tensor of shape `(n_samples, n_targets)`.
+            - Variance tensor of shape `(n_samples, n_targets)`.
+        """
+        if not self.model.supports_uq:
+            msg = f"TransformedEmulator model ({self.model}) does not support UQ."
+            raise RuntimeError(msg)
+        y_pred = self._predict(x, with_grad)
```

Comment:
Is it ever possible that `y_pred` here is a distribution object (e.g., Gaussian) that has a mean and variance and so we do not need to use sampling? I am thinking of cases where we use `TransformedEmulator` without any transforms (or just standardisation) as is the default.

Reply:
This would be great, thanks for the ideas on this. Following our discussion it seems like two options going forward:
- Add analytical support fro `NormalLike` where a `GaussianLike` is transformed to `NormalLike` where each element is an independent gaussian. This would be sufficient maybe for most use cases with mean and variance.
- Look into an analytical transform of the mean when all transforms are bijective
- Look at adding an `analytical: bool` attribute on the `AutoEmulateTransform`

I'll open a new issue to capture additional development on this. It would definitely be great to have a low computational cost path for the mean and variance with gradients when it's possible analytically.